contentcuration/contentcuration/management/commands/exportchannel.py

```diff
@@ -210,25 +211,44 @@ def create_perseus_exercise(ccnode):
         )
         logging.debug("Created exercise for {0} with checksum {1}".format(ccnode.title, assessment_file_obj.checksum))
 
-def create_perseus_zip(ccnode, write_to_path):
-    assessment_items = ccmodels.AssessmentItem.objects.filter(contentnode = ccnode)
 
-    with zipfile.ZipFile(write_to_path, "w") as zf:
+def process_assessment_metadata(ccnode, kolibrinode):
+    # Get mastery model information, set to default if none provided
+    assessment_items = ccnode.assessment_items.all()
+    exercise_data = json.loads(ccnode.extra_fields) if isinstance(ccnode.extra_fields, str) else {}
+    exercise_data = {} if exercise_data is None else exercise_data
+
+    mastery_model = {'type' : exercise_data.get('mastery_model') or exercises.M_OF_N}
+    randomize = exercise_data.get('randomize') or True
+    assessment_item_ids = [a.assessment_id for a in assessment_items]
+
+    if mastery_model['type'] == exercises.M_OF_N:
+        mastery_model.update({'n':exercise_data.get('m') or min(5, assessment_items.count()) or 1})
+        mastery_model.update({'m':exercise_data.get('n') or min(5, assessment_items.count()) or 1})
+
+    exercise_data.update({
+        'mastery_model': mastery_model['type'],
+        'randomize': randomize,
+        'n': mastery_model.get('n'),
+        'm': mastery_model.get('m'),
+        'all_assessment_items': assessment_item_ids,
+        'assessment_mapping': {a.assessment_id : a.type for a in assessment_items},
+    })
+
+    kolibrifilemodel = kolibrimodels.AssessmentMetaData.objects.create(
```

Comment:
create should automatically call save (http://stackoverflow.com/questions/15726553/is-save-called-implicitly-when-calling-create-in-django)